be/src/vec/functions/function_collection_in.h

```diff
@@ -115,36 +118,35 @@ class FunctionStructIn : public IFunction {
             const auto& [col, _] = unpack_if_const(const_column_ptr->column_ptr);
             if (col->is_nullable()) {
                 auto* null_col = vectorized::check_and_get_column<vectorized::ColumnNullable>(col);
-                if (null_col->has_null()) {
+                if (!state->null_in_set && null_col->has_null()) {
```

Comment:
this if else logic maybe not right ! 
think about this in params: (null, xxx, null), the second null will be inserted into args_column_ptr